engine/src/flutter/shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizerTest.mm

```diff
@@ -0,0 +1,119 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/synchronization/waitable_event.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.h"
+#import "flutter/testing/testing.h"
+
+TEST(FlutterThreadSynchronizerTest, NotBlocked) {
+  [FlutterRunLoop ensureMainLoopInitialized];
+
+  FlutterResizeSynchronizer* synchronizer = [[FlutterResizeSynchronizer alloc] init];
+  __block BOOL performed = NO;
+
+  [NSThread detachNewThreadWithBlock:^{
+    [synchronizer performCommitForSize:CGSizeMake(100, 100)
+                                notify:^{
+                                  performed = YES;
+                                }
+                                 delay:0];
+  }];
+
+  CFTimeInterval start = CFAbsoluteTimeGetCurrent();
+
+  while (!performed && CFAbsoluteTimeGetCurrent() - start < 1.0) {
+    [FlutterRunLoop.mainRunLoop pollFlutterMessagesOnce];
+  }
+  EXPECT_EQ(performed, YES);
+}
+
+TEST(FlutterThreadSynchronizerTest, WaitForResize) {
+  [FlutterRunLoop ensureMainLoopInitialized];
+
+  FlutterResizeSynchronizer* synchronizer = [[FlutterResizeSynchronizer alloc] init];
+
+  __block BOOL commit1 = NO;
+  __block BOOL commit2 = NO;
+
+  fml::AutoResetWaitableEvent latch_;
+  fml::AutoResetWaitableEvent& latch = latch_;
```

Comment:
I'm assuming the need for the reference has something to do with usage in a block, and not being able to mark it `__block` for some reason? If so, please add a comment (similar for the one below) -- that's a new one to me!

Reply:
Capturing c++ objects requires copy constructor (which AutoResetWaitableEvent) does not have. What __block does is move this copy to the heap. If still needs copy constructor. Capturing reference on the other hand seems just like capturing a pointer so it works with AutoResetWaitableEvent.